deploy/0-cleanup.sh

```diff
@@ -0,0 +1,25 @@
+#! /bin/bash
+
+set -e
+
+if [ -z "$RESOURCE_GROUP" ]; then
+    echo "RESOURCE_GROUP is not set"
+    exit 1
+fi
+
+# Delete resource group with Key vault, Arc, and all AIO components
+echo "About to delete resource group $RESOURCE_GROUP, please confirm at the next prompt"
+az group delete --name $RESOURCE_GROUP
+
+echo "Resource group $RESOURCE_GROUP deleted, now resetting the K3D cluster to a clean state"
+# Reset K3D cluster
+k3d registry delete devregistry.localhost  
+k3d cluster delete devcluster
+
+# Create local registry for K3D and local development
+k3d registry create devregistry.localhost  --port 5500
```

Comment:
To clean up the K3S cluster and remove all CRDs, pods, namespaces, etc the best and most efficient way is to delete the cluster and recreate it. That is why K3D is quite useful, it has these commands to delete/create entire cluster without having to reinstall K3S.
We could argue the local registry does not need to be deleted/recreated because currently there is no container image pushed to it. But if we end up creating a custom app in the future to show Observability stuff then it's already foreseen.